include/circt/Dialect/FIRRTL/FIRRTLStructure.td

```diff
@@ -253,4 +253,28 @@ def FMemModuleOp : FIRRTLModuleLike<"memmodule"> {
   let hasCustomAssemblyFormat = 1;
 }
 
+def GroupDeclOp : FIRRTLOp<
+  "declgroup",
+  [IsolatedFromAbove, Symbol, SymbolTable, SingleBlock, NoTerminator,
+   HasParent<"firrtl::CircuitOp, firrtl::GroupDeclOp">]
```

Comment:
Edit: I was just cargo-culting this. It seems like a hack that this works even though we're using this in other places. It would be good to switch everything over! Good catch.